helpers/markdownDocumentsReader.ts

```diff
@@ -45,3 +45,21 @@ export function getAllPostDocuments(): PostDocumentWithoutContent[] {
 
   return JSONSerialize(docs);
 }
+
+export function getUpcomingPosts(): PostDocumentWithoutContent[] {
+  const slugs = fs.readdirSync(documentsDirectory);
+  const docs = slugs
+    .map((slug) => getPostDocumentBySlug(slug))
+    .filter((post: PostDocument) => {
+      if (isUpcomingPost(post)) {
+        return true;
+      }
+    })
```

Comment:
```suggestion
    .filter((post: PostDocument) => isUpcomingPost(post))
```
Probably should work